$OpenBSD: patch-bin_youtube-playlists,v 1.1 2013/04/08 19:24:19 sthen Exp $

- fix for playlists with only one item (reported in RT 81068, but not really fixed)

--- bin/youtube-playlists.orig	Mon Apr  1 18:07:46 2013
+++ bin/youtube-playlists	Mon Apr  8 02:06:29 2013
@@ -57,7 +57,9 @@ sub find_video_urls {
     my $urls = [];
     chatty('Parsing XML ... ');
     my $tree = $tpp->parse($xml);
-    for my $entry (@{ $tree->{feed}{entry} || [] }) {
+    my $entries = ref($tree->{feed}{entry}) eq 'ARRAY' ?
+	$tree->{feed}{entry} : [ $tree->{feed}{entry} ];
+    for my $entry (@$entries) {
         my $uri = URI->new($entry->{'media:group'}{'media:player'}{-url});
         $uri->query_param_delete('feature');
         push @$urls, $uri->as_string;
